crates/context/src/lib.rs

```diff
@@ -849,6 +850,43 @@ impl ContextManager {
             bail!("Context not found")
         };
 
+        let this = self.clone();
+
+        let Some(application) = this.get_application(&application_id)? else {
+            bail!("Application with id {:?} not found", application_id)
+        };
+
+        let Some(ContextIdentityValue {
+            private_key: Some(requester_secret),
+        }) = handle.get(&ContextIdentityKey::new(context_id, signer_id))?
+        else {
+            bail!(
+                "Identity {:?} not found in context {:?}",
+                signer_id,
+                context_id
+            )
+        };
+
+        let _ = this
+            .config_client
+            .mutate::<ContextConfigEnv>(
+                this.client_config.new.protocol.as_str().into(),
+                this.client_config.new.network.as_str().into(),
+                this.client_config.new.contract_id.as_str().into(),
```

Comment:
we should read the context config from the `ContextConfig` store record

for example

https://github.com/calimero-network/core/blob/db644002b9093b0363e2af55347ebb849161e07e/crates/context/src/lib.rs#L397-L399

Reply:
updated